src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfFeature.java

```diff
@@ -0,0 +1,1320 @@
+
+    protected final Logger logger = LogManager.getLogger(this.getClass());
+
+    // Metadata fields:
+
+    /**
+     * The relative order of this Feature.
+     * Normally it is a line number indicating the position in the original data file of this feature.
+     */
+    private long                    featureOrderNumber              = -1;
+
+    // Required base GTF Fields:
+    private String                  chromosomeName;
+    private AnnotationSource        annotationSource;
+    private FeatureType             featureType;
+    private int                     genomicStartLocation;
+    private int                     genomicEndLocation;
+    private GenomicStrand           genomicStrand;
+    private GenomicPhase            genomicPhase;
+
+    // "Required" GENCODE GTF Fields:
+    private String                  geneId                          = null;
+    private String                  transcriptId                    = null;
+    private GeneTranscriptStatus    geneStatus                      = null;
+    private GeneTranscriptType      geneType                        = null;
+    private String                  geneName                        = null;
+    private GeneTranscriptType      transcriptType                  = null;
+    private GeneTranscriptStatus    transcriptStatus                = null;
+    private String                  transcriptName                  = null;
+    private int                     exonNumber                      = -1;
+    private String                  exonId                          = null;
+    private LocusLevel              locusLevel                      = null;
+
+    // Optional GENCODE GTF Fields:
+    private ArrayList<OptionalField<?>> optionalFields              = new ArrayList<>();
+
+    // Optional General GTF Fields:
+    private String anonymousOptionalFields                          = null;
+
+    // ================================================================================================
+
+    /**
+     * Populate this GencodeGtfFeature with the given data.
+     */
+    protected GencodeGtfFeature(String[] gtfFields) {
+        chromosomeName          = gtfFields[0];
+        annotationSource        = AnnotationSource.valueOf( gtfFields[1] );
+        featureType             = GencodeGtfFeature.FeatureType.valueOf( gtfFields[2].toLowerCase() );
+        genomicStartLocation    = Integer.valueOf( gtfFields[3] );
+        genomicEndLocation      = Integer.valueOf( gtfFields[4] );
+        genomicStrand           = GenomicStrand.getEnum( gtfFields[6] );
+        genomicPhase            = GenomicPhase.getEnum( gtfFields[7] );
+
+        // Get the extra fields from the last column:
+        String[] extraFields    = gtfFields[8].split(";");
+
+        StringBuilder anonymousOptionalFieldBuilder = new StringBuilder();
+
+        // Now there are "optional" fields to go through (some actually required, some actually optional),
```

Comment:
For these required optional fields I currently do not.  Different versions of the GENCODE GTF files have different required optional fields, so I had not done this yet.  I'll add in some validation.